### ISSUE TITLE
Block Placement

### ISSUE THREAD

**Issue by pinkpig58** [PowerNukkit/PowerNukkit#366]
This issue makes it difficult to place blocks, due to collisions.

### ▶ Steps to Reproduce
Nerdpole or bridge
![75406413-69d87980-58de-11ea-94e6-334c6d73a19e](https://user-images.githubusercontent.com/41856020/87431105-142b1200-c5de-11ea-95c7-6d133c9944b2.gif)

### ✔ Expected Behavior
Blocks should place, just like vanilla, while bridging and nerdpoling.

### ❌ Actual Behavior
Blocks shouldn't disappear while bridging.
The player collides with the block causing it to disappear.

### 💬 Anything else we should know?
This issue was opened for the main Nukkit project, but has not yet been fixed. I'm hoping that PowerNukkit could take a look at this? https://github.com/CloudburstMC/Nukkit/issues/1225

**Comment by joserobjr** [comment]
Some talk about this issue:
https://discordapp.com/channels/728280425255927879/728280425255927882/738750957310967838

**Comment by MrClean1337** [comment]
@joserobjr Fixing that bug will help some pvp servers like my BuildFFA server.
They are crazy in spaming block xd

I found out that this will only happen if you spaming rightclick on mouse.
If you click each block one time, there are no problemes like in this video.
But if you spam, somehow they bug away.

Could be a interact problem

**Comment by pinkpig58** [comment]
**Comment from Nukkit staff member:** the placed block cannot collide with other entities. And nukkit checks that but since client sends the movement packets later than transaction packet we aren't able to perform the correct check. However maybe we could make an exception in case that the player is the only one who collides.

One person reported that removing the following from Level.java was able to help with some of the block placing issues, could this be tested?

```
if (realCount > 0) {
         return null; // Entity in block
}
```

It might also be worth having a look at this too?

```
if (!hand.canPassThrough() && hand.getBoundingBox() != null) {
            Entity[] entities = this.getCollidingEntities(hand.getBoundingBox());
            int realCount = 0;
            for (Entity e : entities) {
                if (e instanceof EntityArrow || e instanceof EntityItem || (e instanceof Player && ((Player) e).isSpectator())) {
                    continue;
                }
                ++realCount;
            }

            if (player != null) {
                Vector3 diff = player.getNextPosition().subtract(player.getPosition());
                if (diff.lengthSquared() > 0.00001) {
                    AxisAlignedBB bb = player.getBoundingBox().getOffsetBoundingBox(diff.x, diff.y, diff.z);
                    if (hand.getBoundingBox().intersectsWith(bb)) {
                        ++realCount;
                    }
                }
            }
```

**Comment by tuilakhanh** [comment]
Fixed in bleeding branch

**Comment by joserobjr** [comment]
Could not reproduce in the `bleeding-1.3.1.2-pn` branch, I think it was fixed by #287 here: https://github.com/PowerNukkit/PowerNukkit/pull/287/files#diff-70123dd1213ca150336e2fda95e45a1aL3228-R3228
